### PR TITLE
test: Run and fix unittests with fatal-warnings

### PR DESCRIPTION
**Pull request by martinpitt**
This is a series of changes to run our unit tests with fatal warnings. Please see individual commit logs for details, I kept unrelated changes separate.

### CODE REVIEW

**Comment by petervo** [comment]
Handling EBADF in cockpitpipe sounds like a good option to me.

**Comment by martinpitt** [comment]
Quick discussion on IRC: Figure out if it's reasonable to tell whether the error came from dbus fd passing, and only then tone it down to a message.

**Comment by martinpitt** [comment]
I reworked the `EBADF` handling in cockpit pipes as discussed. This is explicit and doesn't rely on magic string comparisons.

**Comment by stefwalter** [comment]
Wooot!

Since this is a big change, could you do the following:

```
TEST_OS=fedora-26 test/koji/run-build
```

And check that it passes on all F26 architectures?

File: src/common/cockpitpipe.c [hunk 1]
```diff
@@ -440,6 +441,9 @@ set_problem_from_errno (CockpitPipe *self,
     problem = "access-denied";
   else if (errn == ENOENT || errn == ECONNREFUSED)
     problem = "not-found";
+  /* only warn about Cockpit-internal fds, not opaque user ones */
+  else if (errn == EBADF && self->priv->is_user_fd)
+    problem = "internal-error";
```
Comment (stefwalter) [inline src/common/cockpitpipe.c]: Is this really an "internal-error"? That signifies an internal failure in Cockpit. Do you think another error code better applies? Perhaps "protocol-error" which denotes that the protocol was used wrong? Lets not invent a new code, but I think we may be able to find a better one among the already existing problem codes.

File: src/common/cockpittest.c [hunk 2]
```diff
@@ -220,6 +221,9 @@ cockpit_test_init (int *argc,
   g_assert (g_snprintf (path, sizeof (path), "%s:%s", BUILDDIR, g_getenv ("PATH")) < sizeof (path));
   g_setenv ("PATH", path, TRUE);
 
+  /* For our process and for children */
```
Comment (stefwalter) [inline src/common/cockpittest.c]: The comment is wrong. It's my fault, but it's still wrong :)

File: src/common/cockpittest.c [hunk 3]
```diff
@@ -729,3 +733,22 @@ cockpit_test_find_non_loopback_address (void)
   freeifaddrs (ifas);
   return inet;
 }
+
+void
+cockpit_test_allow_warning (void)
```
Comment (stefwalter) [inline src/common/cockpittest.c]: Shouldn't this be called ```cockpit_test_allow_warnings()```?

File: src/common/cockpittest.c [hunk 4]
```diff
@@ -729,3 +733,22 @@ cockpit_test_find_non_loopback_address (void)
...
+{
+  /* make some noise if this gets called twice */
+  g_return_if_fail (orig_g_debug == NULL);
+  orig_g_debug = g_getenv ("G_DEBUG");
+  g_setenv ("G_DEBUG", "fatal-criticals", TRUE);
+}
+
+void
+cockpit_test_reset_warning (void)
```
Comment (stefwalter) [inline src/common/cockpittest.c]: Ditto ```cockpit_test_reset_warnings()```

File: src/common/cockpitpipe.c (hunk 1 above)
Comment (martinpitt) [inline src/common/cockpitpipe.c]: "protocol-error" or "not-supported" are also close, but neither fits really well.

Comment (martinpitt) [inline src/common/cockpitpipe.c]: Thinking about it again, protocol-error is actually not that bad -- it's not our protocol maybe, but the user's :-)

**Comment by martinpitt** [comment]
I addresses Stef's feedback. Time to toss it to our integration tests, un-bot-ed.

**Comment by martinpitt** [comment]
I locally ran `make check` in a Fedora 26 mock, and now set up koji and submitted a build: https://koji.fedoraproject.org/koji/taskinfo?taskID=18509323 (the source package name is .fc25, not sure why, but it's an F26 build)

**Comment by martinpitt** [comment]
Setting "blocked" until the Koji build finishes.

**Comment by martinpitt** [comment]
https://koji.fedoraproject.org/koji/taskinfo?taskID=18509323 is all green, so unblocking.

**Comment by martinpitt** [comment]
Lots of failures  on Debian: due to `WARNING: cockpit-bridge: dbus-daemon didn't send us a dbus address`, investigating.

**Comment by martinpitt** [comment]
Got it: `dbus-daemon` is not installed in the pbuilder, and `g_spawn_async_with_pipes()` does not detect that immediately. Unfortunately, as we (correctly) launch it without `G_SPAWN_DO_NOT_REAP_CHILD`, we cannot use `kill(pid, 0)` or `waitpid()` or similar to check if it's still actually running, as the forked process still takes a while to iterate through `$PATH`, so in a lot of cases it's still actually running after the read `while` loop. So we could do another loop to wait a bit for it to finish, but this would then slow down production runs with dbus-daemon, and quite pointless.

So I propose to simply downgrade this to `g_message()`, in the spirit that dbus-daemon is an optional component (for bridge) and its absence isn't an internal error. Commit added on top.

**Comment by martinpitt** [comment]
OK, all green now! \o/

**Pull request merged**